actionview/lib/action_view/helpers/form_tag_helper.rb

```diff
@@ -442,8 +443,14 @@ def text_area_tag(name, content = nil, options = {})
       #   check_box_tag 'eula', 'accepted', false, disabled: true
       #   # => <input disabled="disabled" id="eula" name="eula" type="checkbox" value="accepted" />
       def check_box_tag(name, value = "1", checked = false, options = {})
```

Comment:
Hum, yeah default values combined with option hashes are quite problematic.

But then since value has a default too, the same problem can happen with `check_box_tag(name, checked: true)`.

Reply:
> But then since value has a default too, the same problem can happen with `check_box_tag(name, checked: true)`.

Yeah, but at least you get the expected outcome then (checkbox is checked).

Reply:
Would [`extract_options!`](https://api.rubyonrails.org/v7.0/classes/Array.html#method-i-extract_options-21) help in that case?

```suggestion
      def check_box_tag(name, *args)
        options = args.extract_options!
        value, checked = args.empty? ? ["1", false] : args
```

If so, we could document each variant with `:call-seq:`.